src/objectivec/include/ort_genai_objc.h

```diff
@@ -0,0 +1,392 @@
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithModel:(OGAModel*)model
+                                 error:(NSError**)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Set input with NamedTensors type.
+ * @param namedTensors The named tensors.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setInputs:(OGANamedTensors*)namedTensors
+            error:(NSError**)error;
+
+/**
+ * Set input with raw input ids.
+ * @param rawPointer The pointer of the input ids data.
+ * @param inputIdsCount The count of input ids.
+ * @param sequenceLength The sequence length.
+ * @param batchSize The batch size.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setInputIds:(const int32_t*)rawPointer
+      inputIdsCount:(size_t)inputIdsCount
+     sequenceLength:(size_t)sequenceLength
+          batchSize:(size_t)batchSize
+              error:(NSError**)error;
+/**
+ * Set input with sequences type.
+ * @param sequences The sequences.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setInputSequences:(OGASequences*)sequences
+                    error:(NSError**)error;
+
+/**
+ * Set input with name and corresponding tensor.
+ * @param name The input name.
+ * @param tensor The tensor.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setModelInput:(NSString*)name
+               tensor:(OGATensor*)tensor
+                error:(NSError**)error;
+
+/**
+ * Set double option value.
+ * @param key The option key.
+ * @param value The option value.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setSearchOption:(NSString*)key
+            doubleValue:(double)value
+                  error:(NSError**)error;
+/**
+ * Set boolean option value.
+ * @param key The option key.
+ * @param value The option value.
+ * @param error Optional error information set if an error occurs.
+ */
+- (BOOL)setSearchOption:(NSString*)key
+              boolValue:(BOOL)value
+                  error:(NSError**)error;
+@end
+
+/**
+ * The main generator interface that can be used for generation loop.
+ */
+@interface OGAGenerator : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+/**
+ * Creates a generator.
+ *
+ * @param model The model to use.
+ * @param params The generation params to use.
+ * @param error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithModel:(OGAModel*)model
+                                params:(OGAGeneratorParams*)params
+                                 error:(NSError**)error NS_DESIGNATED_INITIALIZER;
+/**
+ * Whether generation is done
+ * @param error Optional error information set if an error occurs.
+ * @return The result, or nil if an error occurs.
+ */
+- (nullable NSNumber*)isDoneWithError:(NSError**)error NS_SWIFT_NAME(isDone());
```

Comment:
This one is tricky. If I use `BOOL`, it will fall into the "BOOL as if the method success" category. So I'm using nullable NSNumber and check it's `boolValue` for success or failure.

Reply:
yeah, that is tricky. does the suggestion in this answer about `swift_error(nonnull_error)` work?
https://stackoverflow.com/a/45566261

it might be nice to have a boolean return value in swift if possible.

Reply:
The attribute `swift_error(nonnull_error)` will handle the naming. But in objective-c, for a signature like `(BOOL)isDoneWithError`,  should I just return `false` to indicate error?

Reply:
alright i'll document this behaviour then. thanks.